src/backend/libc/fs/syscalls.rs

```diff
@@ -638,12 +638,29 @@ unsafe fn utimensat_old(
     target_os = "redox",
     target_os = "wasi",
 )))]
-pub(crate) fn chmodat(dirfd: BorrowedFd<'_>, path: &CStr, mode: Mode) -> io::Result<()> {
-    unsafe { ret(c::fchmodat(borrowed_fd(dirfd), c_str(path), mode.bits(), 0)) }
+pub(crate) fn chmodat(
+    dirfd: BorrowedFd<'_>,
+    path: &CStr,
+    mode: Mode,
+    flags: AtFlags,
+) -> io::Result<()> {
+    unsafe {
+        ret(c::fchmodat(
+            borrowed_fd(dirfd),
+            c_str(path),
+            mode.bits(),
+            flags.bits(),
+        ))
+    }
 }
 
 #[cfg(any(target_os = "android", target_os = "linux"))]
-pub(crate) fn chmodat(dirfd: BorrowedFd<'_>, path: &CStr, mode: Mode) -> io::Result<()> {
+pub(crate) fn chmodat(
+    dirfd: BorrowedFd<'_>,
+    path: &CStr,
+    mode: Mode,
+    _flags: AtFlags,
```

Comment:
This version of `chmodat` should also check the flags and issue an error.

Reply:
Here the libc itself will return the error. It would be wrong to add our own checks, because at any point a future release of glibc should be free to implement the support!

Reply:
Ah, oops… Why did this `linux`/`android` version exist? `libc::fchmodat` is defined right in `src/unix/mod.rs` in libc. Removed to let the actual libc check the flags and issue the error (or emulate the behavior).

Reply:
Interesting, avoiding `/proc` does sound like a good idea!